node-api/handlers/beacon/header.go

```diff
@@ -57,22 +59,27 @@ func (h *Handler) GetBlockHeaderByID(c handlers.Context) (any, error) {
 		c, h.Logger(),
 	)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "failed to bind and validate request")
 	}
 	slot, err := utils.SlotFromBlockID(req.BlockID, h.backend)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to get slot from block ID %s", req.BlockID)
 	}
 	header, err := h.backend.BlockHeaderAtSlot(slot)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to get block header at slot %d", slot)
+	}
+
+	root, err := h.backend.BlockRootAtSlot(slot)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get block root at slot %d", slot)
 	}
 	return beacontypes.NewResponse(&beacontypes.BlockHeaderResponse{
-		Root:      header.GetBodyRoot(),
+		Root:      root, // This is root hash of entire beacon block
 		Canonical: true,
 		Header: &beacontypes.SignedBeaconBlockHeader{
 			Message:   beacontypes.BeaconBlockHeaderFromConsensus(header),
-			Signature: "", // TODO: implement
+			Signature: (&crypto.BLSSignature{}).String(), // TODO: implement
 		},
```

Comment:
Signature needs to be implemented. @berachain/core Any thoughts on this. This is the point that I brought up in the sync call.